.github/workflows/build.yml

```diff
@@ -2,36 +2,102 @@ name: Build
-        run: ./tests/unit/unit
-      - name: Configure (with debug logs)
-        run: make distclean && env CFLAGS='-Werror' ./autogen.sh --enable-debug-messages
-      - name: Build
-        run: make all
-      - name: Configure ARM
-        run: make distclean && env CFLAGS='-Werror' ./autogen.sh --host=arm-linux-gnueabihf --with-only-libndpi
-      - name: Build ARM
-        run: make all
-      - name: Configure Mingw-w64
-        run: make distclean && ./autogen.sh --host=x86_64-w64-mingw32
-      - name: Build Mingw-w64
-        run: make all
+      - name: Install Ubuntu Prerequisites
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt-get update
+          sudo apt-get install autoconf automake libtool pkg-config gettext libjson-c-dev flex bison libpcap-dev
+          sudo apt-get install gcc-arm-linux-gnueabihf gcc-mingw-w64 libc6-dev
+      - name: Install Ubuntu Prerequisites (libgcrypt)
+        if: startsWith(matrix.os, 'ubuntu') && !startsWith(matrix.gcrypt, '--disable-gcrypt')
+        run: |
+          sudo apt-get install libgcrypt20-dev
+      - name: Setup Ubuntu specified compiler (gcc)
+        if: startsWith(matrix.os, 'ubuntu')  && startsWith(matrix.compiler, 'g')
+        run: |
+          sudo apt-get install ${{ format(matrix.compiler, 'cc') }}
+          sudo apt-get install ${{ format(matrix.compiler, '++') }}
+      - name: Setup Ubuntu specified compiler (clang)
+        if: startsWith(matrix.os, 'ubuntu')  && startsWith(matrix.compiler, 'c')
+        run: |
+          sudo apt-get install ${{ format(matrix.compiler, '') }}
+          sudo apt-get install ${{ format(matrix.compiler, '++') }}
+      - name: Installing MacOS prerequisites
+        if: startsWith(matrix.os, 'macOS')
+        run: |
+          brew install autoconf automake libtool pkg-config gettext json-c
+      - name: Install MacOS Prerequisites (libgcrypt)
+        if: startsWith(matrix.os, 'macOS') && !startsWith(matrix.gcrypt, '--disable-gcrypt')
+        run: |
+          brew install libgcrypt
+      - name: Configure nDPI
+        if: startsWith(matrix.compiler, 'default-cc')
+        run: |
+          env CFLAGS='-Werror' ./autogen.sh --enable-debug-messages ${{ matrix.gcrypt }}
+      - name: Configure nDPI with specified GCC version on Ubuntu
+        if: startsWith(matrix.os, 'ubuntu') && startsWith(matrix.compiler, 'g')
+        run: |
+          env CC=${{ format(matrix.compiler, 'cc') }} CXX=${{ format(matrix.compiler, '++') }} CFLAGS='-Werror' ./autogen.sh --enable-debug-messages ${{ matrix.gcrypt }}
+      - name: Configure nDPI with specified CLANG on Ubuntu
+        if: startsWith(matrix.os, 'ubuntu') && startsWith(matrix.compiler, 'c')
+        run: |
+          env CC=${{ format(matrix.compiler, '') }} CXX=${{ format(matrix.compiler, '++') }} CFLAGS='-Werror' ./autogen.sh --enable-debug-messages ${{ matrix.gcrypt }}
+      - name: Build nDPI
+        run: |
+          make all
```

Comment:
Unfortunately, the `all` target does not build *all*.
It is a good idea to build as much as we can, so we have as less as broken code in the repository.
Things that are not build:
```
make -C example ndpiSimpleIntegration
make -C python
```

Not sure if we should also build some of the distro packages e.g. `packages/ubuntu`.

Reply:
One job that also configure's and build's nDPI with `--with-sanitizer`, `--with-pcre` and `--with-maxminddb` might be also interesting. What do you think?